### PR TITLE
LED light effects

### PR DESCRIPTION
**Pull request by rutmanz**
Changes light effects based on game stage:
- Disabled: fire
- Auto: themed BPM
- Teleop: themed rainbow
- Endgame: party rainbow (currently not activated by anything, will add to climber sequence or timer)

### CODE REVIEW

File: src/main/java/org/team1540/robot2022/utils/RevBlinken.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+    public ColorPattern getPattern(Alliance alliance, GameStage stage) {
+        return mappings.get(alliance+""+stage);
+    }
+    
+    public void applyPattern(Alliance alliance, GameStage stage) {
+        ColorPattern pattern = this.getPattern(alliance, stage);
+        this.set(pattern);
+    }
+    /**
```
Comment (natesales) [inline src/main/java/org/team1540/robot2022/utils/RevBlinken.java:61]: Let's reformat the block comments to have a newline above and no newline below

File: src/main/java/org/team1540/robot2022/utils/RevBlinken.java [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+        mappings.put(Alliance.Red+""+GameStage.AUTONOMOUS, LightConstants.redAuto);
+        mappings.put(Alliance.Blue+""+GameStage.AUTONOMOUS, LightConstants.blueAuto);
+
+        mappings.put(Alliance.Red+""+GameStage.TELEOP, LightConstants.redTeleop);
+        mappings.put(Alliance.Blue+""+GameStage.TELEOP, LightConstants.blueTeleop);
+
+        mappings.put(Alliance.Red+""+GameStage.ENDGAME, LightConstants.redEndgame);
+        mappings.put(Alliance.Blue+""+GameStage.ENDGAME, LightConstants.blueEndgame);
+    }
+    /**
```
Comment (natesales) [inline src/main/java/org/team1540/robot2022/utils/RevBlinken.java:34]: \n above

File: src/main/java/org/team1540/robot2022/utils/RevBlinken.java [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+import org.team1540.robot2022.Constants.LightConstants;
+
+import edu.wpi.first.wpilibj.DriverStation.Alliance;
+import edu.wpi.first.wpilibj.motorcontrol.Spark;
+
+
+public class RevBlinken extends Spark {
+    private HashMap<String, ColorPattern> mappings = new HashMap<String, ColorPattern>();
+    /**
```
Comment (natesales) [inline src/main/java/org/team1540/robot2022/utils/RevBlinken.java:13]: \n above

File: src/main/java/org/team1540/robot2022/utils/RevBlinken.java [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+    * Constructor.
```
Comment (natesales) [inline src/main/java/org/team1540/robot2022/utils/RevBlinken.java:14]: Not super descriptive - this should explain what the RevBlinken class does

File: src/main/java/org/team1540/robot2022/utils/RevBlinken.java [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+    *
+    * @param channel The PWM channel that the Blinken is attached to. 0-9 are on-board, 10-19 are on
+    * the MXP port
+    * @throws IndexOutOfBoundsException If the specified PWM channel does not exist.
+    */
+    public RevBlinken(int channel) {
+        super(channel);
+        mappings.put(Alliance.Blue+""+GameStage.DISABLE, LightConstants.disable);
```
Comment (natesales) [inline src/main/java/org/team1540/robot2022/utils/RevBlinken.java:22]: Are these empty strings for inferred casting?

**Review by natesales — CHANGES_REQUESTED** [review]

File: src/main/java/org/team1540/robot2022/utils/RevBlinken.java (hunk 5 above)
Comment (rutmanz) [reply src/main/java/org/team1540/robot2022/utils/RevBlinken.java:22]: Yes, I wasn't sure if there was a easier/cleaner way of doing it

Comment (natesales) [reply src/main/java/org/team1540/robot2022/utils/RevBlinken.java]: :+1: That's clean, just wanted to make sure it was indeed for casting.

**Review by natesales — APPROVED** [review]

**Pull request merged**